Aerospike alpine installation failure

The library fails to install on alpine images.

Steps followed:

```
apk add --no-cache build-base openssl-dev zlib-dev
pip install aerospike
```

```
src/main/client/set_xdr_filter.c: In function 'AerospikeClient_SetXDRFilter':
      src/main/client/set_xdr_filter.c:119:2: error: unknown type name 'uint'; did you mean 'int'?
        119 |  uint request_length = strlen(fmt_str) + strlen(data_center_str_p) +
            |  ^~~~
            |  int
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]
```

uint is not recognised by gcc in alpine linux

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -55,7 +55,7 @@ jobs:
 
           git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
           cd aerospike-client-python
-          git checkout ${{ env.GITHUB_REF_NAME }}
+          git checkout $GITHUB_REF_NAME
           git submodule update --recursive
 
           python3 -m pip install build

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -55,7 +55,7 @@ jobs:
 
           git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
           cd aerospike-client-python
-          git checkout ${GITHUB_REF##*/}
+          git checkout ${{ env.GITHUB_REF_NAME }}
           git submodule update --recursive
 
           python3 -m pip install build

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -55,9 +55,9 @@ jobs:
 
           git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
           cd aerospike-client-python/
-          echo $GITHUB_REF_NAME
-          git checkout $GITHUB_REF_NAME
-          # git submodule update --recursive
+          echo ${{ github.ref_name }}
+          git checkout ${{ github.ref_name }}
+          git submodule update --recursive
 
           # python3 -m pip install build
           # python3 -m build

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -41,7 +41,7 @@ jobs:
       run: sleep 3
 
     - name: Get server ip
-      run: SERVER_IP=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' aerospike) >> $GITHUB_ENV
+      run: echo SERVER_IP=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' aerospike) >> $GITHUB_ENV
 
     # Use this to get current branch of workflow
     - name: Inject slug/short variables
@@ -85,6 +85,11 @@ jobs:
           sed -i "s/hosts: 127.0.0.1:3000/hosts: $SERVER_IP:3000/" config.conf
 
           python3 -m pytest new_tests/
+
+    - name: Cleanup
+      run: |
+        docker container stop aerospike
+        docker container rm aerospike
 
   build:
     runs-on: ubuntu-latest

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -39,6 +39,8 @@ jobs:
         image: alpine:latest
         run: |
           apk add py3-pip
+          apk add python3-dev
+          apk add zlib-dev
           apk add git
           # C client dependencies
           apk add automake

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -54,17 +54,18 @@ jobs:
           apk add graphviz # (for make docs)
 
           git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
-          cd aerospike-client-python
+          cd aerospike-client-python/
+          echo $GITHUB_REF_NAME
           git checkout $GITHUB_REF_NAME
-          git submodule update --recursive
+          # git submodule update --recursive
 
-          python3 -m pip install build
-          python3 -m build
-          python3 -m pip install dist/*.whl
-          cd test/
-          # requirements.txt is not working atm
-          python3 -m pip install pytest
-          python3 -m pytest new_tests/
+          # python3 -m pip install build
+          # python3 -m build
+          # python3 -m pip install dist/*.whl
+          # cd test/
+          # # requirements.txt is not working atm
+          # python3 -m pip install pytest
+          # python3 -m pytest new_tests/
 
   build:
     runs-on: ubuntu-latest

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -37,6 +37,7 @@ jobs:
     - uses: addnab/docker-run-action@v3
       with:
         image: alpine:latest
+        options: -e GITHUB_REF_NAME=${{ github.ref_name }}
         run: |
           apk add py3-pip
           apk add python3-dev
@@ -55,8 +56,7 @@ jobs:
 
           git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
           cd aerospike-client-python/
-          echo ${{ github.ref_name }}
-          git checkout ${{ github.ref_name }}
+          git checkout $GITHUB_REF_NAME
           git submodule update --recursive
 
           # python3 -m pip install build

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -34,10 +34,8 @@ jobs:
   alpine:
     runs-on: ubuntu-latest
     steps:
-    - name: Set up Aerospike Database
-      uses: reugn/github-action-aerospike@dev
-      with:
-        port: 3000
+    - name: Run Aerospike server
+      run: docker run -d --name aerospike -p 3000-3002:3000-3002 aerospike/aerospike-server
 
     - name: Wait for database to be ready
       run: sleep 3
@@ -84,7 +82,7 @@ jobs:
           python3 -m pip install pytest
 
           # Configure tests
-          sed -i 's/hosts: 127.0.0.1:3000/hosts: ${SERVER_IP}:3000/' config.conf
+          sed -i "s/hosts: 127.0.0.1:3000/hosts: $SERVER_IP:3000/" config.conf
 
           python3 -m pytest new_tests/
 

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -31,6 +31,15 @@ jobs:
         check-path: 'src'
         clang-format-version: 14
 
+  alpine:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: addnab/docker-run-action@v3
+      with:
+        image: alpine:latest
+        run: |
+          uname -a
+
   build:
     runs-on: ubuntu-latest
     strategy:

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -34,10 +34,16 @@ jobs:
   alpine:
     runs-on: ubuntu-latest
     steps:
+    # Use this to get current branch of workflow
+    - name: Inject slug/short variables
+      uses: rlespinasse/github-slug-action@v4
+      with:
+        short-length: 8 # Same as v3 and before
+
     - uses: addnab/docker-run-action@v3
       with:
         image: alpine:latest
-        options: -e GITHUB_REF_NAME=${{ github.ref_name }}
+        options: -e GITHUB_REF_NAME=${{ env.GITHUB_REF_NAME }}
         run: |
           apk add py3-pip
           apk add python3-dev

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -55,6 +55,9 @@ jobs:
 
           git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
           cd aerospike-client-python
+          git checkout ${GITHUB_REF##*/}
+          git submodule update --recursive
+
           python3 -m pip install build
           python3 -m build
           python3 -m pip install dist/*.whl

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -38,7 +38,28 @@ jobs:
       with:
         image: alpine:latest
         run: |
-          uname -a
+          apk add py3-pip
+          apk add git
+          # C client dependencies
+          apk add automake
+          apk add make
+          apk add musl-dev
+          apk add gcc
+          apk add openssl-dev
+          apk add lua-dev
+          apk add libuv-dev  # (for node.js)
+          apk add doxygen  # (for make docs)
+          apk add graphviz # (for make docs)
+
+          git clone --recurse-submodules https://github.com/aerospike/aerospike-client-python
+          cd aerospike-client-python
+          python3 -m pip install build
+          python3 -m build
+          python3 -m pip install dist/*.whl
+          cd test/
+          # requirements.txt is not working atm
+          python3 -m pip install pytest
+          python3 -m pytest new_tests/
 
   build:
     runs-on: ubuntu-latest

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -34,6 +34,17 @@ jobs:
   alpine:
     runs-on: ubuntu-latest
     steps:
+    - name: Set up Aerospike Database
+      uses: reugn/github-action-aerospike@dev
+      with:
+        port: 3000
+
+    - name: Wait for database to be ready
+      run: sleep 3
+
+    - name: Get server ip
+      run: SERVER_IP=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' aerospike) >> $GITHUB_ENV
+
     # Use this to get current branch of workflow
     - name: Inject slug/short variables
       uses: rlespinasse/github-slug-action@v4
@@ -43,7 +54,7 @@ jobs:
     - uses: addnab/docker-run-action@v3
       with:
         image: alpine:latest
-        options: -e GITHUB_REF_NAME=${{ env.GITHUB_REF_NAME }}
+        options: -e GITHUB_REF_NAME=${{ env.GITHUB_REF_NAME }} -e SERVER_IP=${{ env.SERVER_IP }}
         run: |
           apk add py3-pip
           apk add python3-dev
@@ -65,13 +76,17 @@ jobs:
           git checkout $GITHUB_REF_NAME
           git submodule update --recursive
 
-          # python3 -m pip install build
-          # python3 -m build
-          # python3 -m pip install dist/*.whl
-          # cd test/
-          # # requirements.txt is not working atm
-          # python3 -m pip install pytest
-          # python3 -m pytest new_tests/
+          python3 -m pip install build
+          python3 -m build
+          python3 -m pip install dist/*.whl
+          cd test/
+          # requirements.txt is not working atm
+          python3 -m pip install pytest
+
+          # Configure tests
+          sed -i 's/hosts: 127.0.0.1:3000/hosts: ${SERVER_IP}:3000/' config.conf
+
+          python3 -m pytest new_tests/
 
   build:
     runs-on: ubuntu-latest

--- a/BUILD.md
+++ b/BUILD.md
@@ -58,6 +58,27 @@ Dependencies:
  sudo pacman -S binutils gcc
  ```
 
+### Alpine Linux
+
+Dependencies:
+
+```sh
+apk add py3-pip
+apk add python3-dev
+apk add zlib-dev
+apk add git
+# C client dependencies
+apk add automake
+apk add make
+apk add musl-dev
+apk add gcc
+apk add openssl-dev
+apk add lua-dev
+apk add libuv-dev  # (for node.js)
+apk add doxygen  # (for make docs)
+apk add graphviz # (for make docs)
+```
+
 ### macOS
 
 By default macOS will be missing command line tools.

--- a/README.rst
+++ b/README.rst
@@ -13,7 +13,7 @@ Compatibility
 
 The Python client for Aerospike works with Python 3.7, 3.8, 3.9, and 3.10 running on
 **64-bit** macOS 10.15+ and Linux (RHEL/CentOS 7 & 8; Debian 8, 9 & 10; Ubuntu
-18.04 & 20.04).
+18.04 & 20.04; Alpine Linux).
 
 **NOTE:** Aerospike Python client 5.0.0 and up MUST be used with Aerospike server 4.9 or later.
 If you see the error "-10, ‘Failed to connect’", please make sure you are using server 4.9 or later.

--- a/src/main/client/set_xdr_filter.c
+++ b/src/main/client/set_xdr_filter.c
@@ -117,8 +117,8 @@ PyObject *AerospikeClient_SetXDRFilter(AerospikeClient *self, PyObject *args,
 
     // - 6 for format char
     unsigned int request_length = strlen(fmt_str) + strlen(data_center_str_p) +
-                          strlen(namespace_str_p) + strlen(base64_filter) + 1 -
-                          6;
+                                  strlen(namespace_str_p) +
+                                  strlen(base64_filter) + 1 - 6;
     request_str_p = cf_malloc(request_length * sizeof(char));
     if (request_str_p == NULL) {
         as_error_update(&err, AEROSPIKE_ERR_CLIENT,

--- a/src/main/client/set_xdr_filter.c
+++ b/src/main/client/set_xdr_filter.c
@@ -116,7 +116,7 @@ PyObject *AerospikeClient_SetXDRFilter(AerospikeClient *self, PyObject *args,
     }
 
     // - 6 for format char
-    uint request_length = strlen(fmt_str) + strlen(data_center_str_p) +
+    unsigned int request_length = strlen(fmt_str) + strlen(data_center_str_p) +
                           strlen(namespace_str_p) + strlen(base64_filter) + 1 -
                           6;
     request_str_p = cf_malloc(request_length * sizeof(char));

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -196,28 +196,28 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
 
     // Don't need to copy lua script to user path if it already exists there
     bool same_lua_script_at_user_path = false;
-    if (access(copy_filepath, F_OK) == 0){
+    if (access(copy_filepath, F_OK) == 0) {
         struct stat original_fstat, copied_fstat;
         stat(filename, &original_fstat);
         stat(copy_filepath, &copied_fstat);
-        if(original_fstat.st_ino == copied_fstat.st_ino){
+        if (original_fstat.st_ino == copied_fstat.st_ino) {
             same_lua_script_at_user_path = true;
         }
     }
 
-    if (!same_lua_script_at_user_path){
+    if (!same_lua_script_at_user_path) {
         // Copy lua script to user path
         copy_file_p = fopen(copy_filepath, "w+");
-        if(copy_file_p == NULL){
+        if (copy_file_p == NULL) {
             as_error_update(&err, AEROSPIKE_ERR_CLIENT,
                             "No permissions to write lua file to user path");
             goto CLEANUP;
         }
 
         char ch;
-        while((ch = fgetc(file_p)) != EOF){
+        while ((ch = fgetc(file_p)) != EOF) {
             int retVal = fputc(ch, copy_file_p);
-            if(retVal == EOF){
+            if (retVal == EOF) {
                 as_error_update(&err, AEROSPIKE_ERR_CLIENT,
                                 "Write of lua file to user path failed");
                 goto CLEANUP;
@@ -229,7 +229,7 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
 
     // Copy lua script to buffer so we can send it to server
     int numBytesRead = fread(buff, 1, fileSize, file_p);
-    if(numBytesRead != fileSize){
+    if (numBytesRead != fileSize) {
         as_error_update(&err, AEROSPIKE_ERR_CLIENT,
                         "Unable to send lua file to server");
         goto CLEANUP;

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -200,10 +200,10 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
         // Don't need to copy lua script to user path if it already exists there
         bool same_lua_script_at_user_path = false;
         if (access(copy_filepath, F_OK) == 0){
-            struct stat *original_fstat, *copied_fstat;
-            fstat(filename, original_fstat);
-            fstat(copy_filepath, copied_fstat);
-            if(original_fstat->st_ino == copied_fstat->st_ino){
+            struct stat original_fstat, copied_fstat;
+            fstat(filename, &original_fstat);
+            fstat(copy_filepath, &copied_fstat);
+            if(original_fstat.st_ino == copied_fstat.st_ino){
                 same_lua_script_at_user_path = true;
             }
         }

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -215,6 +215,8 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
                             "No permissions to write lua file to user path");
             goto CLEANUP;
         }
+
+        char ch;
         while((ch = fgetc(file_p)) != EOF){
             int retVal = fputc(ch, copy_file_p);
             if(retVal == EOF){
@@ -227,13 +229,11 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
     }
 
     // Copy lua script to buffer so we can send it to server
-    while((ch = fgetc(file_p)) != EOF){
-        int retVal = fputc(ch, copy_file_p);
-        if(retVal == EOF){
-            as_error_update(&err, AEROSPIKE_ERR_CLIENT,
-                            "Unable to send lua file to server");
-            goto CLEANUP;
-        }
+    int numBytesRead = fread(buff, 1, fileSize, file_p);
+    if(numBytesRead != fileSize){
+        as_error_update(&err, AEROSPIKE_ERR_CLIENT,
+                        "Unable to send lua file to server");
+        goto CLEANUP;
     }
 
     as_bytes_init_wrap(&content, bytes, size, true);

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -249,7 +249,7 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
                       &content);
     Py_END_ALLOW_THREADS
     if (err.code != AEROSPIKE_OK) {
-        as_error_update(&err, err.code, NULL);
+        // as_error_update(&err, err.code, NULL);
         goto CLEANUP;
     }
     else {

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -201,8 +201,8 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
         bool same_lua_script_at_user_path = false;
         if (access(copy_filepath, F_OK) == 0){
             struct stat original_fstat, copied_fstat;
-            fstat(filename, &original_fstat);
-            fstat(copy_filepath, &copied_fstat);
+            stat(filename, &original_fstat);
+            stat(copy_filepath, &copied_fstat);
             if(original_fstat.st_ino == copied_fstat.st_ino){
                 same_lua_script_at_user_path = true;
             }

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -196,49 +196,44 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
 
     uint8_t *buff = bytes;
 
-    if (access(self->as->config.lua.user_path, W_OK) == 0) {
-        // Don't need to copy lua script to user path if it already exists there
-        bool same_lua_script_at_user_path = false;
-        if (access(copy_filepath, F_OK) == 0){
-            struct stat original_fstat, copied_fstat;
-            stat(filename, &original_fstat);
-            stat(copy_filepath, &copied_fstat);
-            if(original_fstat.st_ino == copied_fstat.st_ino){
-                same_lua_script_at_user_path = true;
-            }
+    // Don't need to copy lua script to user path if it already exists there
+    bool same_lua_script_at_user_path = false;
+    if (access(copy_filepath, F_OK) == 0){
+        struct stat original_fstat, copied_fstat;
+        stat(filename, &original_fstat);
+        stat(copy_filepath, &copied_fstat);
+        if(original_fstat.st_ino == copied_fstat.st_ino){
+            same_lua_script_at_user_path = true;
         }
+    }
 
-        if(!same_lua_script_at_user_path){
-            copy_file_p = fopen(copy_filepath, "w+");
-            if (copy_file_p) {
-                int read = (int)fread(buff, 1, LUA_FILE_BUFFER_FRAME, file_p);
-                if (read && fwrite(buff, 1, read, copy_file_p)) {
-                    while (read) {
-                        size += read;
-                        buff += read;
-                        read = (int)fread(buff, 1, LUA_FILE_BUFFER_FRAME, file_p);
-                        if (!fwrite(buff, 1, read, copy_file_p)) {
-                            break;
-                        }
-                    }
-                }
-                else {
-                    as_error_update(&err, AEROSPIKE_ERR_CLIENT,
-                                    "Write of lua file to user path failed");
-                    goto CLEANUP;
-                }
-            }
-            else {
+    if (!same_lua_script_at_user_path){
+        // Copy lua script to user path
+        copy_file_p = fopen(copy_filepath, "w+");
+        if(copy_file_p == NULL){
+            as_error_update(&err, AEROSPIKE_ERR_CLIENT,
+                            "No permissions to write lua file to user path");
+            goto CLEANUP;
+        }
+        while((ch = fgetc(file_p)) != EOF){
+            int retVal = fputc(ch, copy_file_p);
+            if(retVal == EOF){
                 as_error_update(&err, AEROSPIKE_ERR_CLIENT,
                                 "Write of lua file to user path failed");
                 goto CLEANUP;
             }
         }
+        fseek(file_p, 0, SEEK_SET);
     }
-    else {
-        as_error_update(&err, AEROSPIKE_ERR_CLIENT,
-                        "No permissions to write lua file to user path");
-        goto CLEANUP;
+
+    // Copy lua script to buffer so we can send it to server
+    while((ch = fgetc(file_p)) != EOF){
+        int retVal = fputc(ch, copy_file_p);
+        if(retVal == EOF){
+            as_error_update(&err, AEROSPIKE_ERR_CLIENT,
+                            "Unable to send lua file to server");
+            goto CLEANUP;
+        }
     }
 
     as_bytes_init_wrap(&content, bytes, size, true);

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -192,8 +192,6 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
         goto CLEANUP;
     }
 
-    int size = 0;
-
     uint8_t *buff = bytes;
 
     // Don't need to copy lua script to user path if it already exists there
@@ -225,6 +223,7 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
                 goto CLEANUP;
             }
         }
+        // Reset source fd so we can copy lua script to buffer
         fseek(file_p, 0, SEEK_SET);
     }
 
@@ -236,7 +235,7 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
         goto CLEANUP;
     }
 
-    as_bytes_init_wrap(&content, bytes, size, true);
+    as_bytes_init_wrap(&content, bytes, fileSize, true);
 
     // Invoke operation
     Py_BEGIN_ALLOW_THREADS

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -197,18 +197,35 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
     uint8_t *buff = bytes;
 
     if (access(self->as->config.lua.user_path, W_OK) == 0) {
+        // Don't need to copy lua script to user path if it already exists there
+        bool same_lua_script_at_user_path = false;
+        if (access(copy_filepath, F_OK) == 0){
+            struct stat *original_fstat, *copied_fstat;
+            fstat(filename, original_fstat);
+            fstat(copy_filepath, copied_fstat);
+            if(original_fstat->st_ino == copied_fstat->st_ino){
+                same_lua_script_at_user_path = true;
+            }
+        }
 
-        copy_file_p = fopen(copy_filepath, "w+");
-        if (copy_file_p) {
-            int read = (int)fread(buff, 1, LUA_FILE_BUFFER_FRAME, file_p);
-            if (read && fwrite(buff, 1, read, copy_file_p)) {
-                while (read) {
-                    size += read;
-                    buff += read;
-                    read = (int)fread(buff, 1, LUA_FILE_BUFFER_FRAME, file_p);
-                    if (!fwrite(buff, 1, read, copy_file_p)) {
-                        break;
+        if(!same_lua_script_at_user_path){
+            copy_file_p = fopen(copy_filepath, "w+");
+            if (copy_file_p) {
+                int read = (int)fread(buff, 1, LUA_FILE_BUFFER_FRAME, file_p);
+                if (read && fwrite(buff, 1, read, copy_file_p)) {
+                    while (read) {
+                        size += read;
+                        buff += read;
+                        read = (int)fread(buff, 1, LUA_FILE_BUFFER_FRAME, file_p);
+                        if (!fwrite(buff, 1, read, copy_file_p)) {
+                            break;
+                        }
                     }
+                }
+                else {
+                    as_error_update(&err, AEROSPIKE_ERR_CLIENT,
+                                    "Write of lua file to user path failed");
+                    goto CLEANUP;
                 }
             }
             else {
@@ -216,11 +233,6 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
                                 "Write of lua file to user path failed");
                 goto CLEANUP;
             }
-        }
-        else {
-            as_error_update(&err, AEROSPIKE_ERR_CLIENT,
-                            "Write of lua file to user path failed");
-            goto CLEANUP;
         }
     }
     else {

--- a/src/main/client/udf.c
+++ b/src/main/client/udf.c
@@ -243,7 +243,6 @@ PyObject *AerospikeClient_UDF_Put(AerospikeClient *self, PyObject *args,
                       &content);
     Py_END_ALLOW_THREADS
     if (err.code != AEROSPIKE_OK) {
-        // as_error_update(&err, err.code, NULL);
         goto CLEANUP;
     }
     else {

